### DISCUSSION TITLE
How to use text predictor to deal with a text classification problem with imbalanced positive and negative samples of 1:10?

### DISCUSSION THREAD

**Discussion by bioinformatica** [autogluon/autogluon#2371] [Q&A]

**Comment by rxjx** [comment]
I've used text predictor to solve exactly these kinds of problems. IME, the usual tricks for imbalanced data aren't really necessary at this level of imbalance as long as you have at least a couple of hundred examples of the minority class. The trick is to have good diversity of your examples and decent coverage over your know distribution. I do wonder whether some sort of prior specification would be helpful. That would definitely be a feature request I'd push for.

**Reply by bioinformatica** [reply]
It is a dataset with a positive-negative ratio of 1500 : 15000. I tried upsampling and downsampling, although recall increased from 50% to 80%, but precision dropped from 80% to 30%, and roc_auc dropped from 92 to 90.

**Reply by rxjx** [reply]
I've definitely been able to get good predictors with that amount and proportion of data without any additional techniques. I'd guess it's more to do with your data than autogluon itself.
